lib/IRGen/GenMeta.cpp

```diff
@@ -4431,7 +4431,11 @@ llvm::Value *irgen::emitHeapMetadataRefForHeapObject(IRGenFunction &IGF,
     return emitLoadOfHeapMetadataRef(IGF, object,
                                      getIsaEncodingForType(IGF.IGM, objectType),
                                      suppressCast);
-
+  // Is it an opened class existential?
+  if (objectType.isAnyClassReferenceType() && isa<ArchetypeType>(objectType))
```

Comment:
@rjmccall John, is this a proper way to check that emitLoadOfHeapMetadataRef could be used for class existentials? Or do I need to add something else, e.g. to check if it is not a tagged pointer or something like this?